block_util/src/raw_sync.rs

```diff
@@ -65,15 +60,22 @@ impl AsyncIo for RawFileSync {
         iovecs: Vec<libc::iovec>,
         user_data: u64,
     ) -> AsyncIoResult<()> {
-        read_vectored_sync(
-            offset,
-            iovecs,
-            user_data,
-            &mut self.raw_file,
-            &self.eventfd,
-            &mut self.completion_list,
-            &mut self.semaphore,
-        )
+        let result = unsafe {
```

Comment:
It will be nice to see this lock removed:
```
-   15.21%     0.44%  _disk0_q0        cloud-hypervisor    [.] <block_util::raw_sync::RawFileSync as block_util::async_io::AsyncIo>::read_vectored                                                                                                                                                                           ▒
   - 14.77% <block_util::raw_sync::RawFileSync as block_util::async_io::AsyncIo>::read_vectored                                                                                                                                                                                                                              ▒
      + 7.92% block_util::read_vectored_sync                                                                                                                                                                                                                                                                                 ▒
        2.13% cfree@GLIBC_2.2.5                                                                                                                                                                                                                                                                                              ▒
        1.12% __pthread_mutex_lock                                                                                                                                                                                                                                                                                           ▒
        0.71% __libc_write                                                                                                                                                                                                                                                                                                   ▒
        0.68% __pthread_mutex_unlock_usercnt                                                                                                                                                                                                                                                                                 ▒
        0.67% <qcow::raw_file::RawFile as std::io::Seek>::seek
```